### PR TITLE
Dde stop being default

### PR DESCRIPTION
**Pull request by AnyhowStep**
In reference to Redmine ticket 12195,

> Problem:
> The Data Team Helper currently assumes all instruments are DDE -- it shows incorrect Incomplete numbers for instruments that were never defined as DDE instruments.
> Solution:
> Check the Config table for which instruments are DDE-enabled, and adapt query accordingly.

I honestly have no idea if the attempted fix will do anything useful. I tried to follow the outlined fix on that ticket. Basically, I check if the test is a DDE or not and filter out results that aren't.

But I don't know a single thing about this module so I don't know if it's had an impact. Could someone who knows what's going on do the testing and point out what cases it isn't covering? Or if it's doing something else completely?

### CODE REVIEW

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 1]
```diff
@@ -420,6 +420,11 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
         $visit_label=null,
         $candidate=null
     ) {
+        $config         =& NDB_Config::singleton();
+        $ddeInstruments = $config->getSetting('DoubleDataEntryInstruments');
+        for ($i=0; $i<count($ddeInstruments); ++$i) {
```
Comment (christinerogers) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: Given count($ddeInstruments) properly returns 0 for an unset/empty array i.e. when no DDE instruments are returned from Config settings, this looks fine.

File: modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc [hunk 2]
```diff
@@ -430,9 +435,21 @@ class NDB_Form_Data_Team_Helper extends NDB_Form
...
+        if (count($ddeInstruments) > 0) {
+            $ddeInstruments = implode(",", $ddeInstruments);
+            $where         .= "
+                AND (f.test_name IN ($ddeInstruments) OR
+                f.commentid NOT LIKE 'DDE_%')
+            ";
+        } else {
+            $where .= "
+                AND (f.commentid NOT LIKE 'DDE_%')
+            ";
```
Comment (christinerogers) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: lgtm

Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: I think this logic looks right to me, but I'm not really familiar with how the module uses the query either.

Can someone who knows the module test this before we merge it?

Comment (driusan) [inline modules/data_team_helper/php/NDB_Form_data_team_helper.class.inc]: @kongtiaowang since there haven't been any volunteers, can you test this?

**Comment by sruthymathew123** [comment]
@AnyhowStep Please rebase

**Comment by sruthymathew123** [comment]
This change is already incorporated in IBIS with the changes I made with data team helper. I am familiar with the module and tested it.

**Comment by AnyhowStep** [comment]
Wait, why does this need rebase?

**Pull request merged**